nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java

```diff
@@ -244,9 +246,17 @@ public static SolrClient createSolrClient(final PropertyContext context, final S
         params.set(HttpClientUtil.PROP_MAX_CONNECTIONS, maxConnections);
         params.set(HttpClientUtil.PROP_MAX_CONNECTIONS_PER_HOST, maxConnectionsPerHost);
 
+        // CloudSolrClient.Builder now requires a List of ZK addresses and znode for solr as separate parameters
+        String zk[] = solrLocation.split("/");
+        final List zkList = Arrays.asList(zk[0].split(","));
+        String zkRoot = "/";
+        if (zk.length > 1 && ! zk[1].isEmpty()) {
+            zkRoot += zk[1];
+        }
+

```

Comment:
I think we should move this ZK parsing block inside the else statement on line 274 where we know the Solr Type is CLOUD, just to be on the safer side since we wouldn't need to do this parsing when the type is STANDARD.

If you agree I can make this change as part of the merge.

Reply:
@bbende thanks for the review. Yes you are right we should move it to else statement in 274. You can make the change as part of the merge. Thanks!